src/app/(with-main-header)/boards/[id]/update/page.tsx

```diff
@@ -3,16 +3,85 @@
 import Button from "@/components/common/Button";
 import Icon from "@/components/common/Icon";
 import AddPhotoAlternateRoundedIcon from "@mui/icons-material/AddPhotoAlternateRounded";
-import PostImage from "../../PostImage";
 import Input from "@/components/common/Input";
-import { FormEvent, useState } from "react";
+import { FormEvent, useEffect, useState } from "react";
+import { useParams, useRouter } from "next/navigation";
+import { boardApi } from "@/api/board/board";
+import { userApi } from "@/api/user/user";
+import CloseRoundedIcon from "@mui/icons-material/CloseRounded";
 
 export default function BoardsUpdate() {
+  const params = useParams();
+  const router = useRouter();
+  const boardId = Number(params.id);
+
   const [title, setTitle] = useState("");
+  const [context, setContext] = useState("");
+  const [images, setImages] = useState<File[]>([]);
+  const [existingImages, setExistingImages] = useState<string[]>([]);
+  const [deletedImages, setDeletedImages] = useState<string[]>([]);
+
+  useEffect(() => {
+    const checkAuthAndFetchBoard = async () => {
+      try {
+        const [userResponse, boardResponse] = await Promise.all([
+          userApi.getCurrentUserInfo(),
+          boardApi.getBoardByBoardId(boardId),
+        ]);
+
+        if (!boardResponse) {
+          router.push("/boards");
+          return;
+        }
+
+        if (userResponse?.data.userId !== boardResponse?.data.userId) {
+          alert("수정 권한이 없습니다.");
+          router.push(`/boards/${boardId}`);
```

Comment:
```suggestion
          router.replace(`/boards/${boardId}`);
```
push로 할 경우에 사용자가 뒤로가기 시, 다시 이 페이지로 와서 리디렉션 되기 때문에 replace가 더 좋은 것 같습니다!